RUFAS/input_manager.py

```diff
@@ -789,3 +792,273 @@ def flush_pool(self) -> None:
+
+        Notes
+        -----
+        This function is designed to verify the existence of specified properties
+        within the metadata of a particular variable. It returns a boolean indicating
+        the existence of the properties, and a KeyError in case of missing metadata
+        or properties.
+
+        Parameters
+        ----------
+        variable_name : str
+            The name of the variable for which the metadata is to be checked.
+        properties_blob_key : str
+            The key representing the specific properties blob in the metadata to check.
+
+        Returns
+        -------
+        tuple
+            A tuple containing a boolean and an exception:
+            - The boolean is True if the properties exist, False otherwise.
+            - KeyError if the metadata is not loaded or the properties are not found,
+              otherwise None.
+        """
+        info_map = {"class": self.__class__.__name__,
+                    "function": self._metadata_properties_exists.__name__,
+                    }
+        if not self.__metadata:
+            om.add_error("No metadata loaded", "No metadata is loaded to the InputManager.", info_map)
+            return False, ValueError("No metadata loaded.")
+        if properties_blob_key not in self.__metadata["properties"]:
+            om.add_error("No metadata found", f"No metadata is found for variable '{variable_name}' with given "
+                                              f"properties_blob_key {properties_blob_key}.", info_map)
+            return False, KeyError(f"No metadata is found for variable '{variable_name}' with given properties_blob_key"
+                                   f" {properties_blob_key}.")
+        return True, None
+
+    def _add_variable_to_pool(self, variable_name: str, data: Dict[str, Any], properties_blob_key: str,
+                              eager_termination: bool, is_dict_variable: bool) -> (bool, ValueError | None):
+        """
+        Adds a variable to the pool after validating its data against specified metadata properties.
+
+        Notes
+        -----
+        This function processes and validates the input data for a variable based on
+        its metadata properties. It then adds the validated data to a pool. The function
+        also provides an option for eager termination in case of invalid data.
+
+        Parameters
+        ----------
+        variable_name : str
+            The name of the variable to be added to the pool.
+        data : Dict[str, Any]
+            The data associated with the variable that needs validation and addition to the pool.
+        properties_blob_key : str
+            The key in the metadata properties against which the data is validated.
+        eager_termination : bool
+            Flag indicating whether the function should return early in case of invalid data.
+        is_dict_variable : bool
```

Comment:
```suggestion
        is_variable_dict : bool
```
We don't say `Is developer Allister`, we say `is Allister a developer`